internal/node/hybrid/ip_validator_test.go

```diff
@@ -0,0 +1,170 @@
+package hybrid_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/eks-hybrid/internal/api"
+	"github.com/aws/eks-hybrid/internal/aws/eks"
+	"github.com/aws/eks-hybrid/internal/node/hybrid"
+)
+
+func TestHybridNodeProvider_ValidateNodeIP(t *testing.T) {
+	tests := []struct {
+		name                string
+		nodeConfig          *api.NodeConfig
+		remoteNetworkConfig *eks.RemoteNetworkConfig
+		wantErr             bool
```

Comment:
i would suggest making this a string so you can actually expect the error message (or partial error message)